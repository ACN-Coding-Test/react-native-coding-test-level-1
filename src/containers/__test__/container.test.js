import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';

import Container from '../Container';

test('renders Loading correctly', () => {
    const tree = renderer
        .create(
            <Container
                backgroundColor="white"
                statusBarColor="red"
                title="PokeDex"
                isMain={true}>
                <View style={{flex: 1}}></View>
            </Container>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
