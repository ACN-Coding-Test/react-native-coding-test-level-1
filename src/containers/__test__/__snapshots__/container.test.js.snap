// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "height": 750,
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "height": 70,
      }
    }
    testID="header"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": "5%",
          "paddingRight": "5%",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="back-button"
        >
          <Text
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    {
                      "letterSpacing": 1.5,
                    },
                  ],
                ],
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flex": 3,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "letterSpacing": 1.5,
              "textAlign": "center",
            }
          }
        >
          PokeDex
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={false}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          {
            "flex": 1,
            "paddingBottom": "5%",
            "paddingTop": "5%",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
